examples/step-48/step-48.cc

```diff
@@ -222,60 +222,63 @@ namespace Step48
   //@sect4{SineGordonOperation::apply}
 
   // This function performs the time stepping routine based on the cell-local
-  // strategy. First the destination vector is set to zero, then the cell-loop
-  // is called, and finally the solution is multiplied by the inverse mass
-  // matrix. The structure of the cell loop is implemented in the cell finite
+  // strategy. Note that we need to set the destination vector to zero before
+  // we add the integral contributions of the current time step (via the
+  // FEEvaluation::distribute_local_to_global() call). In this tutorial, we
+  // let the cell-loop do the zero operation via the fifth `true` argument
+  // passed to MatrixFree::cell_loop. The loop can schedule the zero operation
+  // closer to the operations on vector entries for supported vector entries,
+  // thereby possibly increasing data locality (the vector entries that first
+  // get zeroed are later re-used in the `distribute_local_to_global()`
+  // call). The structure of the cell loop is implemented in the cell finite
   // element operator class. On each cell it applies the routine defined as
   // the <code>local_apply()</code> method of the class
   // <code>SineGordonOperation</code>, i.e., <code>this</code>. One could also
-  // provide a function with the same signature that is not part of a class.
+  // provide a function with the same signature that is not part of a
+  // class. Finally, the result of the integration is multiplied by the
+  // inverse mass matrix.
   template <int dim, int fe_degree>
   void SineGordonOperation<dim, fe_degree>::apply(
     LinearAlgebra::distributed::Vector<double> &                     dst,
     const std::vector<LinearAlgebra::distributed::Vector<double> *> &src) const
   {
-    dst = 0;
-    data.cell_loop(&SineGordonOperation<dim, fe_degree>::local_apply,
-                   this,
-                   dst,
-                   src);
+    data.cell_loop(
+      &SineGordonOperation<dim, fe_degree>::local_apply, this, dst, src, true);
     dst.scale(inv_mass_matrix);
   }
 
 
+
   //@sect3{Equation data}
 
   // We define a time-dependent function that is used as initial
   // value. Different solutions can be obtained by varying the starting
   // time. This function has already been explained in step-25.
   template <int dim>
-  class ExactSolution : public Function<dim>
+  class InitialCondition : public Function<dim>
```

Comment:
Its not that important, but I would prefer to keep the old name so that we match with step-25.

Reply:
Actually, I disagree. If the class doesn't represent the solution at all times, then `ExactSolution` is a misleading name. And if it's only used for the initial conditions, then that's how it should be called.

The original sin would in that case be with step-25 :-(

Reply:
Alright, that makes it a 2:1 decision to go to the `InitialCondition` name here, because this is not the solution for all times. I should add that step-25 actually derives an `InitialCondition` based on the analytic solution, so it is not a complete departure at this point. In particular since the current setup would need a different function in 2D/3D, which I was simply too lazy to do. I will add a reference to step-25 where the reader could find these things.

Reply:
Fair enough.